mlir/test/python/dialects/linalg/ops.py

```diff
@@ -606,3 +606,38 @@ def tensor_pack(src, dst):
         # CHECK:           return %[[VAL_4]] : tensor<128x128xf32>
         # CHECK:         }
         print(module)
+
+
+@run
+def test_infer_contraction_dimensions():
+    with Context(), Location.unknown():
+        module = ir.Module.parse(
+            r"""
+            module {
+            func.func @matmul(%arg0: tensor<4x4xf32>, %arg1: tensor<4x4xf32>)
+                -> tensor<4x4xf32> {
+                %cst = arith.constant 0.0 : f32
+                %0 = linalg.fill ins(%cst : f32) outs(%arg0 : tensor<4x4xf32>) -> tensor<4x4xf32>
+                %1 = linalg.matmul ins(%arg0, %arg1 : tensor<4x4xf32>, tensor<4x4xf32>)
+                outs(%0 : tensor<4x4xf32>) -> tensor<4x4xf32>
+                return %1 : tensor<4x4xf32>
+            }
+            }
+            """
+        )
+        func_op = module.body.operations[0]
```

Comment:
Do we need to check that the module parsed? I'm thinking about when the IR format changes in the future and someone has to update this.

Reply:
welll really we shouldn't be using textual IR here (though I don't understand figuring out the corresponding python is tough...)

Reply:
you can find some examples at [mlir/test/python/dialects/linalg/ops.py](https://github.com/makslevental/llvm-project/blob/a72616de18c0814ad37b5748d6bdc60b825dd889/mlir/test/python/dialects/linalg/ops.py#L43-L47)